pyqt-apps/siriushla/widgets/scrn_view.py

```diff
@@ -0,0 +1,828 @@
+
+        self.label = QLabel(')', self)
+        self.label.setMaximumSize(10, 40)
+        layout.addWidget(self.label, 7, 5)
+
+        # - Theta
+        self.label_Theta = QLabel('Theta', self)
+        self.label_Theta.setAlignment(Qt.AlignCenter)
+        self.label_Theta.setMaximumHeight(40)
+        layout.addWidget(self.label_Theta, 8, 1, 1, 5)
+        self.label = QLabel('(', self)
+        self.label.setMaximumSize(10, 40)
+        layout.addWidget(self.label, 9, 1)
+        self.PyDMLabel_ThetaDimFei = PyDMLabel(
+            parent=self,
+            init_channel='ca://'+self.prefix+self.device+':ThetaDimFei-Mon')
+        self.PyDMLabel_ThetaDimFei.setAlignment(Qt.AlignHCenter)
+        layout.addWidget(self.PyDMLabel_ThetaDimFei, 9, 2, 1, 3)
+        self.PyDMLabel_ThetaNDStats = PyDMLabel(
+            parent=self,
+            init_channel='ca://'+self.prefix+self.device+':ThetaNDStats-Mon')
+        self.PyDMLabel_ThetaNDStats.setAlignment(Qt.AlignHCenter)
+        self.PyDMLabel_ThetaNDStats.setVisible(False)
+        layout.addWidget(self.PyDMLabel_ThetaNDStats, 9, 2, 1, 3)
+        self.label = QLabel(')', self)
+        self.label.setMaximumSize(10, 40)
+        layout.addWidget(self.label, 9, 5)
+
+        layout.addItem(
+            QSpacerItem(40, 20, QSzPlcy.Expanding, QSzPlcy.Minimum), 10, 2)
+        return layout
+
+    def _handleShowStatistics(self, visible):
+        self.PyDMLabel_CenterXDimFei.setVisible(not visible)
+        self.PyDMLabel_CenterXNDStats.setVisible(visible)
+        self.PyDMLabel_CenterYDimFei.setVisible(not visible)
+        self.PyDMLabel_CenterYNDStats.setVisible(visible)
+        self.PyDMLabel_ThetaDimFei.setVisible(not visible)
+        self.PyDMLabel_ThetaNDStats.setVisible(visible)
+        self.PyDMLabel_SigmaXDimFei.setVisible(not visible)
+        self.PyDMLabel_SigmaXNDStats.setVisible(visible)
+        self.PyDMLabel_SigmaYDimFei.setVisible(not visible)
+        self.PyDMLabel_SigmaYNDStats.setVisible(visible)
+
+    def _saveCalibrationGrid(self):
+        Thread(target=self._saveCalibrationGrid_thread, daemon=True).start()
+
+    def _saveCalibrationGrid_thread(self):
+        roi_h = self.PyDMLabel_ROIHeight.text()
+        roi_w = self.PyDMLabel_ROIWidth.text()
+        roi_offsetx = self.PyDMLabel_ROIOffsetX.text()
+        roi_offsety = self.PyDMLabel_ROIOffsetY.text()
+        self.PyDMSpinbox_ROIHeight.value_changed(
+                self.image_view.image_maxheight)
+        self.PyDMSpinbox_ROIHeight.send_value()
+        self.PyDMSpinbox_ROIWidth.value_changed(
+                self.image_view.image_maxwidth)
+        self.PyDMSpinbox_ROIWidth.send_value()
+        self.PyDMSpinbox_ROIOffsetX.value_changed(0)
+        self.PyDMSpinbox_ROIOffsetX.send_value()
+        self.PyDMSpinbox_ROIOffsetY.value_changed(0)
+        self.PyDMSpinbox_ROIOffsetY.send_value()
+        time.sleep(0.3)
+        self.image_view.saveCalibrationGrid()
+        time.sleep(0.3)
```

Comment:
Do you think it is needed to wait here? I think we could drop this...

Reply:
I agree, I will drop this!